Meta data protocol v3 (#51)
worldbank/dime-standards: markdown-scheme_v3.Rmd

---
title: "Metadata Template"
author: "Christian Krestel"
date: "2023-06-27"
output:
  pdf_document:
    includes:
      in_header: "preamble.tex"
    toc: true
    toc_depth: 3
    latex_engine: xelatex
---
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Version number: 3

# Template: Code repository catalogue

1. Create new project: Type script

> 2. Screenshot of the \emph{complete} first page

Reviewer: ```suggestion
2. Screenshot of the \emph{complete} title page
```

> 3. For a published paper, prepare the link to the repository. For a WP prepare a zip that contains all data and code; For \textbf{DIME verified} packages also the reproducibility verification statement

Reviewer: ```suggestion
3. For a published paper, prepare the link to the repository. For a WP prepare a zip that contains: readme, code files, all public data files, data statement (if nto included in readme) and license. For \textbf{DIME verified} packages also the reproducibility verification statement
```

# Information on metadata
This is information about the creator of the metadata. Sufficient fields
\begin{enumerate}

>   \item Name: Second name only: Smith

Reviewer: ```suggestion
  \item Name: Surname only: Smith
```

>   \item Abbreviation: PS

Reviewer: ```suggestion
  \item Abbreviation: First and Last initials: PS
```

>   \item Name: Smith

Author: I changed the order of the item. It should be there only once

>   \item Name: Smith

Reviewer: i see that now, i was looking at the pdf you'd sent by email.

  \item Affiliation: World Bank
  \item Production date: Year-Month-Day

>   \item Version number: Entered as 1,2,3

Reviewer: clarify whether this is metadata version or reproducibility package version.

Author: For meta data

Reviewer: cool, that should be noted in the protocol

\end{enumerate}

# Project description
## Title statement
Sufficient fields are
\begin{enumerate}

>   \item Primary ID: Ideally a DOI or an organization consistent scheme;

Reviewer: agreed this would not be a DOI so this needs to be revised

      \begin{enumerate}
        \item Published Paper: 
            \begin{enumerate}
              \item A corresponding PRWP exists: FORMAT
                \begin{lstlisting}

>                   PP_CountryCodeIso3_Year_PRWP-PRWPno_versionno

Reviewer: would be useful to include link to country ISO codes or put in annex to the protocol

Reviewer: is the versionno here the version of the catalog entry or the PRWP? clarify

Author: Ok I will add a reference to the ISO country codes. This is the version number of the PRWP

Reviewer: cool, please specify in protocol that it's version number of PRWP

                  Example: PP_CRI_2018_PRWP-8524_v01
                \end{lstlisting}

>               \item A corresponding PRWP does not exit: FORMAT

Reviewer: ```suggestion
              \item A corresponding PRWP does not exist: FORMAT
```

                \begin{lstlisting}
                  PP_CountryCodeIso3_Year_EXT-EXTno_versionno
                  Example: PP_CRI_2018_EXT-1_v01
                \end{lstlisting}

>                     EXTno is a unique running number assigned by Dime.

Reviewer: ```suggestion
                    EXTno is a unique running number assigned by the metadata producer.
```

we should create a spreadsheet for recording these numbers and link to it from the protocol

Author: We should.

            \end{enumerate}

>         \item Working Paper: FORMAT

Reviewer: ```suggestion
        \item Policy Research Working Paper: FORMAT
```

          \begin{lstlisting}

>             RR_CountryCodeIso3_Year_PRWP-PRWPno_versionno

Reviewer: out of curiousity, why RR and not WP?

Author: I believe that perhaps Olivier was using this

Reviewer: ok, we are not tied to that, should come up with whatever convention is best for this use case.

            RR_CRI_2018_PRWP-8525_v01
          \end{lstlisting}
        \end{enumerate}

>   \item Other identifiers: DOI and https://doi.org/number

Reviewer: Just a note that we still need to work out the exact protocol for generating and entering the DOIs for packages that we publish directly. DECDG owns the account for generating DOIs). they use the metadata from NADA to generate the DOI, so the sequence will, i think, need to be that we create the catalog entry, share the draft with someone at DG, they run a script to harvest the metadata and create the DOI, and then they come back to us with the DOI (or directly enter it into the catalogue). I've raised the issue with the DG team and will update, so let's leave this flagged as requiring a later update for now.

Author: Ok. We have a primary ID, the DOI can be added later

  \item Title: Of the following form
    \begin{lstlisting}

>       Reproducibility package for TITEL(paper)

Reviewer: ```suggestion
      Reproducibility package for TITLE (full title of the paper)
```

    \end{lstlisting}
  \item Subtitle:
    \begin{itemize}
      \item WP: World Bank Policy Research Working Paper (PRWP) number
      \item PP: DOI number (no https)
    \end{itemize}
\end{enumerate}

## Authors and contributors
Sufficient are
\begin{enumerate}

>     \item Authoring entity: author, affiliation and ORCID number and email when available; For world bank authors use \textbf{World Bank}. Example:

Reviewer: ```suggestion
    \item Authoring entity: author, affiliation and ORCID number and email when available. For World Bank authors use \textbf{World Bank} and always provide email address (query in staff directory as needed). Example:
```

      \begin{verbatim}
        Christoph Lakner World Bank ORCID 0000, clakner@worldbank.org
      \end{verbatim}

>     \item Contributors: like RAs (not conference participants). Also coeditors if specifically mentioned. FORMAT

Reviewer: it's not clear what the format is demonstrating here

Author: It is an example of all the necessary items and that we use World Bank and not The World Bank

    \begin{lstlisting}
      Coeditor
    \end{lstlisting}
    \item Sponsors: All financial contributions and individual/general grants

>     \item Acknowledgements: Usually empty; Example: Corporate Taxation: special mentioning ministry employees for data assistance (could go to acknowledgment statement)

Reviewer: suggest removing, or making a much clearer case for what goes here that would not be included in acknowledgement statement.

Author: OK. This is ambiguous. Perhaps best directly not to use this field.

    \item Acknowledgement Statement:
      \begin{verbatim}
        We are grateful to Benu Bidani, Francisco Ferreira, Luis-Felipe..
      \end{verbatim}
      \begin{verbatim}
        .. and seminar participants at the World Bank’s Poverty ...
      \end{verbatim}
  \end{enumerate}

## Version statement
Sufficient are
\begin{enumerate}
    \item Project completion date: FORMAT: YEAR-MONTH
      \begin{itemize}
        \item PP: Date at which the article was published: Year-Month

>         \item WP: \textcolor{red}{Must be supplied, bc it is needed for the front page

Reviewer: If the WP is already published, this can be extracted from the cover page. if the WP is not already published, suggest we assume it will be published in the same month of the review.

        of cataloge}
      \end{itemize}
    \item Version date: FORMAT: YEAR-MONTH-DAY
      \begin{itemize}
        \item PP: empty

>         \item WP: data of publication as research paper in WB Year-Month-Day

Reviewer: not clear to me how this is different from the 'project completion date' explained above. when would these be different? make a clearer case for how the information is distinct or drop one.

      \end{itemize}
\end{enumerate}

## Scope and Coverage
Sufficient are
\begin{enumerate}

>     \item Abstract: Abstract of the article; \textcolor{red}{Hyphens are a problem. Check all abstracts. Also carriage returns are a problem, direct copy pasting leads to bad results}

Reviewer: is copy-pasting a problem when one pastes as plain text? are problems resolved by pasting into Word first and then pasting to the Editor? "Check all abstracts" --> when are the problems visible, as soon as you paste or only when published? include a bit more detail here.

    \item Geographic areas: Format: Name, Code, Type
    \begin{lstlisting}
      Lebanon, LBN, ISO-3166 Alpha-3
    \end{lstlisting}

>     \item Keywords: No keywords if none are given; Yes if given.

Reviewer: what does given mean here? like if they are provided in the metadata for the paper? i don't think PRWP ever has these, so this is probably only relevant to journal articles.

Author: And AEA journals also do  not have them.

Reviewer: then suggest removing, JEL codes is sufficent

    \item JEL codes:
      \begin{verbatim}
        ID D63
        Name Equity, Justice, Inequality
        Parent ID (empty)
        Vocabulary JEL Classifications
        URL https://www.aeaweb.org/econlit/jelCodes.php?view=jel
      \end{verbatim}
\end{enumerate}

## Processes and Output

> The following fields exist

Reviewer: want to focus on minimum required set of fields not what fields exist

Author: Yes, but here we have more possibilities, this is why I added the description for the items we are not currently using, which is Process, Institutional Review board, approval process and Reviews and comments. In theory, if I understand the description correctly, one could describe the lifecycle from start to end of the process. We do not use this yet. It is not obvious how we could use it.

Reviewer: for now we want to focus on establishing the minimum necessary fields that all catalogue entries must have. it's fine if other fields are included, but those should be clearly marked as optional (and unless they are clearly useful i would be inclined not to include in our template ...)

\begin{enumerate}
    \item Process: empty for PP, tbd for WP
    \item Institutional review board: empty
    \item Approval process: empty
    \item Reviews and comments:
      \begin{itemize}
        \item WP: If available (tbd, 2023-06-22)
        \item PP: empty

>       \end{itemize}

Reviewer: it's not clear to me why any of these should be required, can you make a case?

Author: Not really. But I also was only dealing with published papers. One would have to ask the people submitting working papers if this were something useful. I personally do not think it is useful, unless we have something like in the previous comment, a process that we describe. For papers and working papers it does not seem to be useful.

Reviewer: i don't think it's useful. let's omit for our protocol.

    \item Output: 
        \begin{itemize}
          \item Type    
            \begin{itemize}
              \item WP: Working Paper
              \item PP: Published Paper (not paper, not article, not Article)
            \end{itemize}
          \item Title: Title of the paper

>           \item Authors: Authors: all comma separated; first name, last name. Patty Smith, Patty Smith, Petty Smith

Reviewer: isn't this repeating the information from authoring entity? why do we need it in both places?

Author: We did it like this when Olivier showed me the editor. I do not mind it: True we replicate the information, but it is very nice to have all information on the output drop down

            \begin{lstlisting}[breaklines]
                Toulouse School of Economics Working Papers No 1391
            \end{lstlisting}
          \item DOI: YES (if it exists)
          \item URL: YES (Editor displays an error at https, but it is accepted)
        \end{itemize}
    \item Language: English, EN
\end{enumerate}

## Data
Sufficient are
\begin{enumerate}
    \item Data statement:\footnote{ Description: An overall statement on the data used in the project. A spearate field is provided to list and document the origin and key characteristics of the datasets}. For external packages, the following rule:
        \begin{itemize}
          \item If all data is public:
            \begin{itemize}
            \item Fill in: "All data is public and contained in the (external)         reproducibility package."
            \item Add data availability statement if available from journal package; often separate from the zip, has to be copied over manually.
            \end{itemize}
          \item If some data is confidential: Paste from the readme notes about confidentiality.
        \end{itemize}

>     \item Datasets:\footnote{Description of this field: "This field is used to provide an itemized list of datasets used in the project. The data are not documented here (specific metadata are available for documenting data of different types, like the DDO for microdata, the ISO 19139 for geographic datasets etc)"} Not used for PP, will be used for WP, if they data sets are of proper value and stored in another WB repository (like microdata library). \textcolor{red}{No such case arose so far. TBD}

Reviewer: certainly applies for PPs. i would think more common for PPs to have data published separately and called from rep package (e.g., data are catalogued elsewhere and referenced in the data statement). for WPs we may be directly involved in cataloguing so that is different.

Author: I will change the wording

        \begin{enumerate}
          \item Name
          \item Data set ID: (empty)
          \item Note: (empty)
          \item Access policy:
            \begin{itemize}
              \item Published elsewhere: Note access/license
              \item Published with package: Authors must confirm there is no confidentiality.
            \end{itemize}
          \item License (empty)
          \item License URL (empty)
          \item Data URL
        \end{enumerate}
\end{enumerate}

## Methods, software and scripts

> Fields that are not listed can be discarded.

Reviewer: i presume this is true in all sections?

\begin{enumerate}

>   \item Methods: Econometric methods used. \textcolor{red}{This field will learn from past entries.} Possibly add notes (more details about the context)

Reviewer: let's have a working list that we either include directly here or include a link to (probably better to link to a doc that can be separately updated as this is one where i think frequent updates would be required in the early phases of the pilot)

>   \item Software: Stata 17.0, R, etc

Reviewer: suggest separating out software and version, so that we can have a simple filter by language (but we certainly don't want to lose the version information, just to store it separately)

Author: It is stored separately in the meta data

  \item Scripts:
      \begin{enumerate}

>           \item File name (WP+PP): Name of the zip file. Naming convention: PrimaryID\_prg\_v01.zip

Reviewer: (WP+PP) - in other parts of the protocol, i assumed that fields that are not specified as either WP or PP apply to both. confusing to start to label (WP+PP) here; either need to do it consistently across full protocol or not at all. my suggestion would be to default that the protocol applies to all and specify when there are deviations.

Reviewer: naming convention: why "_prg"?

Author: For program. When adding external resources, I cannot choose package, which would be more fitting. The closest fit was program. Thus for consistency

Reviewer: got it.

          \item Zip package (WP+PP): Same as file name\footnote{File name and zip package are identical, because we do not list files separately, for which the field file name is there.}
          \item Title: 
            \begin{lstlisting}[breaklines]
              Reproducibility package (data and code) for Title
            \end{lstlisting}
            Title may be shortened if adequate.

>           \item Authors: WP+PP: empty\footnote{According to descr: A list of authors and coauthors of the script} \textcolor{red}{If authors of the paper and the code differ and authors want to indicate this, add here list of authors for the code!}

Reviewer: why is this in red? seems like a useful explanation to keep in the protocol. why is there a footnote description for this field?

Author: This was to remind me that this information is new (at least it was a couple of days ago)

>           \item Date: Year-Month (usually the completion/version date)

Reviewer: completion/version data of the paper?

Author: Yes

          \item Software: Stata

>           \item Description: empty

Reviewer: I would think this should be a description of what's in the zip?

Author: We can do that, its a decision to be made. It is though a lot of work and it is not immediately clear what is the value of this. Usually the readme has a summary anyway.

Reviewer: I think it can be a simple description, like a tree of what's in the zip since people can't see it til they download (e.g. the package contains a readme file, a data folder with 5 datasets, a scripts folder with a main script and 8 sub-scripts, etc)

>           \item Methods: \textcolor{red}{This is methods used in the script}. Usually empty

Reviewer: need to be clear on whether we are entering methods once (in part 1 of this section of the metadata) or at script level. i think it does make sense to only do once per paper. but in that case we wouldn't need this field in our template.

Author: Indeed. At this specific point it would only make sense to have something at the script level, I completely agree.

>           \item Dependencies: Here enter all packages necessary:\footnote{It may be preferable to enter this in Software, libraries: \textcolor{red}{Error: Must be an array}. Does not accept more than one entry, and also not a string. Behavior of the editor his random here}

Reviewer: this sounds like something that should be flagged with mehmood (suggest creating an issue on this behavior). can flag that this part of the protocol needs to be updated after the behavior is resolved.

Author: Ok. Can add this to his github. Will check if this also happens with the catalogue and the instance of the editor at qa. Also, it worked for a while and then started to stop working.

Reviewer: that is strange, please do document in an issue

            \begin{lstlisting}
              coefplot, distinct, lincomeest, lprobust.ado
            \end{lstlisting}
          \item Instructions: Write: "See readme in reproducibility package"
          \item Repository: ICPSR, Harvard Dataverse, Journal webpage etc. For ICPSR
                the correct description is

>                 \begin{lstlisting}

Reviewer: these examples are not just ICPSR need to adjust text above

                  AEA Data and Code Repository (ICPSR)
                  Reproducible Research Repository (World Bank)
                \end{lstlisting}

>           \item Notes: \textcolor{red}{Here who verified the package}. 

Reviewer: I think we may want a specific field for reproducibility verification, we should discuss with Olivier about adding this. Notes field is a fine workaround for now.

            \begin{lstlisting}[breaklines]
            Checked for reproducibility externally at the journal.
            Checked for reproducibility at Development Impact Evaluation (DIME), World Bank
            \end{lstlisting}
          \item License:
            \begin{itemize}
              \item PP
              \begin{itemize}
                \item ICPSR:
                  \begin{lstlisting}[breaklines]
                  Creative Commons Attribution 4.0 International (CC BY 4.0) License
                  https://creativecommons.org/licenses/by/4.0/
                  \end{lstlisting}
              \end{itemize}

>               \item WP: Modified BSD3

Reviewer: add url here

            \end{itemize}
      \end{enumerate}
  \item Repository: Harvard, ICPSR- Type empty - URL, Journal webpage
    \begin{lstlisting}[breaklines]
        AEA Data and Code Repository
        https://www.openicpsr.org/openicpsr/aea
    \end{lstlisting}
  \item Reproducibility:
      \begin{enumerate}

>             \item Technology environment: empty

Reviewer: should not be empty, should be the environment used in the rep check, this should be noted in our report

Author: Will split this up into PP and WP

            \item Technology Requirements: Standard, Cluster...; include runtime if available
            \item Reproduction instructions: No details:
                \begin{lstlisting}[breaklines]
                 A readme file with detailed instructions is part of the (external) reproducibility package.
                \end{lstlisting}
                 Short details:
                  \begin{lstlisting}
                    A readme file and a single do file, AP_final_AEJ. do are provided. 
                  \end{lstlisting}
        \end{enumerate}
\end{enumerate}

## Access and Rights
\begin{enumerate}

>     \item Disclaimer: \textcolor{red}{Perhaps not necessary}

Reviewer: not necessary. discussing with Legal now what we need for terms of use of the site but we can remove this from the metadata.

      \begin{lstlisting}[breaklines]
        The views expressed here are the authors, and do not necessarily reflect the official views of the World Bank, its Executive Directors, or the countries they represent.
      \end{lstlisting}
    \item Confidentiality: YES or NO, or PARTIAL. \textcolor{red}{Do not add description}
    \item Citation requirement: (empty)

>     \item License: as in scripts

Reviewer: do we need to repeat this here?

Author: You have it in the data statement. But it is important information, so perhaps an isolated clear field has some value here.

Reviewer: ok, agree there seems to be value in repeating

        \begin{itemize}
              \item PP
              \begin{itemize}
                \item ICPSR:
                  \begin{lstlisting}[breaklines]
                  Creative Commons Attribution 4.0 International (CC BY 4.0) License
                  https://creativecommons.org/licenses/by/4.0/
                  \end{lstlisting}
              \end{itemize}
              \item WP: 
              \begin{lstlisting}
                Modified BSD3
                https://opensource.org/license/bsd-3-clause/
              \end{lstlisting}
            \end{itemize}

>     \item Copyright: (empty)

Reviewer: do we need this field? don't include fields that will always be empty

\end{enumerate}

## Contacts
\begin{enumerate}

>     \item WB author or corresponding author: Fill fields Name, email, affiliation, role Author

Reviewer: must list WB author. also list corresponding author if different.

      \begin{lstlisting}
      Patty Smith
      Author
      psmith@worldbank.org
      World Bank
      \end{lstlisting}
    \item Reproducibility group contact
      \begin{lstlisting}
        Reproducibility WBG
        reproducibility@worldbank.org
        reproducibility.worldbank.org
      \end{lstlisting}
\end{enumerate}

# Tags
\begin{enumerate}

>   \item Tag: Partially Reproducible, Fully Reproducible, Not Reproducible, Tag group Reproducible

Reviewer: we will not publish any packages that are not reproducible. but perhaps you were thinking of code only? perhaps we could do the following: 
- Fully reproducible: public code and data
- Partially reproducible (restricted data): public code, at least some restricted-access data
- Partially reproducible (restricted code): public data, at least some restricted-access code

Author: Is this not too much for just tags

  \item Type: PRWP, Published Paper
\end{enumerate}

# External Resources
\begin{enumerate}
  \item Zip file/link for PP and WP
  \begin{enumerate}
    \item File type: prg
    \item Title
      \begin{lstlisting}

>         Reproducibility package (data and code) for TITLE (of paper), PID PrimaryID

Reviewer: i think confusing to specify "data and code" here. keep the title the same as elsewhere.

Author: It is called as such also in outputs

      \end{lstlisting}
      Title of paper may be shortened\footnote{It is not possible to put the PID into the subtitle as we do with the title at the beginning, because the subtitle is not displayed. Putting the PID into the subtitle means dropping the PID}
    \item Date: equal to version date/completion date: FORMAT Year-Month
    \item Description: 
    \begin{itemize}
      \item WP: empty
      \item PP:
        \begin{lstlisting}
          Externally available on ICPSR
        \end{lstlisting}
    \end{itemize}
    \item Resource attachment: 
    \begin{itemize}
      \item WP: Filename in the format PID\_prg\_v01.zip (as the zipname in scripts)
      \item PP: Link to the repository/article
    \end{itemize}
  \end{enumerate}
  \item For packages checked by DIME, attach the reproducibility verification statement. Title: Reproducibility verification certificate. 

>   \item All else emtpy.

Reviewer: no need to include
